### PR TITLE
make ldap testable, add travis testing

### PR DESCRIPTION
**Pull request by Habbie**

### CODE REVIEW

**Comment by Habbie** [comment]
Review + merge welcome

File: regression-tests/backends/ldap-master [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		cat > pdns-ldap.conf << __EOF__
+module-dir=./modules
+launch=ldap
+ldap-basedn=$LDAPBASEDN
+ldap-binddn=$LDAPUSER
+ldap-secret=$LDAPPASSWD
+ldap-method=$layout
+__EOF__
+
+		# $PDNSUTIL --config-dir=. --config-name=oracle import-tsig-key test $ALGORITHM $KEY
```
Comment (pieterlexis) [inline regression-tests/backends/ldap-master]: These lines can go?

Comment (Habbie) [inline regression-tests/backends/ldap-master]: removed

File: regression-tests/backends/ldap-master [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+		layout=${context:5}
+		ldapdelete -D $LDAPUSER -w $LDAPPASSWD -r $LDAPBASEDN || true
+		ldapadd -D $LDAPUSER -w $LDAPPASSWD << __EOF__
+dn: $LDAPBASEDN
+objectClass: organizationalUnit
+$LDAPRELATTR
+__EOF__
+
+		../pdns/zone2ldap --dnsttl=yes --basedn=$LDAPBASEDN --layout=$layout --named-conf=named.conf | ldapmodify -D $LDAPUSER -w $LDAPPASSWD -c > /dev/null || true
```
Comment (pieterlexis) [inline regression-tests/backends/ldap-master]: maybe we should add a `$ZONE2LDAP` var for this... (like `$PDNSUTIL`)

Comment (Habbie) [inline regression-tests/backends/ldap-master]: good idea

**Comment by pieterlexis** [comment]
It could use some words in `modules/ldapbackend/README.md` on why certain tests are skipped?

**Comment by Habbie** [comment]
doc issues noted in #3358

File: .travis.yml [hunk 3]
```diff
@@ -111,6 +111,21 @@ before_script:
   - echo -e "[pdns-sqlite3-1]\nDriver = SQLite3\nDatabase = ${PWD}/regression-tests/pdns.sqlite3\n\n[pdns-sqlite3-2]\nDriver = SQLite3\nDatabase = ${PWD}/regression-tests/pdns.sqlite32\n" > ${HOME}/.odbc.ini
   - export GODBC_SQLITE3_DSN=pdns-sqlite3-1
 
+  # ldap-backend test setup
+  - sudo apt-get -qq --no-install-recommends install slapd ldap-utils
```
Comment (pieterlexis) [inline .travis.yml]: could the spaces be replaced by new-lines? this makes .travis.yml better diffable.

File: .travis.yml [hunk 4]
```diff
@@ -111,6 +111,21 @@ before_script:
...
+  - pushd /tmp/ldap-dns
+  - for schema in /etc/ldap/schema/{core,cosine}.schema ${TRAVIS_BUILD_DIR}/modules/ldapbackend/{dnsdomain2,pdns-domaininfo}.schema ; do echo include $schema ; done > ldap.conf
+  - mkdir slapd.d
+  - slaptest -f ldap.conf -F slapd.d
+  - sudo cp slapd.d/cn=config/cn=schema/cn={*dns*.ldif /etc/ldap/slapd.d/cn=config/cn=schema/
+  - sudo chown -R openldap:openldap /etc/ldap/slapd.d/
+  - sudo service slapd restart
+  - popd
+  - sudo mkdir /var/lib/ldap/powerdns
+  - sudo chown openldap:openldap /var/lib/ldap/powerdns
```
Comment (pieterlexis) [inline .travis.yml]: if `/var/lib/ldap` is already owned by `openldap`, these 2 lines could be rolled into `sudo -u openldap mkdir /var/lib/ldap/powerdns`

File: .travis.yml (hunk 3 above)
Comment (Habbie) [inline .travis.yml]: done

File: .travis.yml (hunk 4 above)
Comment (Habbie) [inline .travis.yml]: done, with -p for robustness

**Comment by Habbie** [comment]
All nits covered I think. When green I will squash unless more comments come in.

**Comment by Habbie** [comment]
Squashed.

**Comment by Habbie** [comment]
Squashed and green, ready for final review (if felt necessary) and merge.

**Pull request merged**